### ISSUE TITLE
Is it possible to share clipboard with host?

### ISSUE THREAD

**Issue by euforic** [lima-vm/lima#424]
Currently I am trying to share my clipboard between host and vm to allow for easier copy and pasting.

**Comment by AkihiroSuda** [comment]
Clipboard of what?

**Comment by euforic** [comment]
An example would be if I copied something from vim (in vm) then wanted to paste into safari (on my Mac).

**Comment by afbjorklund** [comment]
Are you using something other than the regular terminal application, to access your `limactl shell` ?

**Comment by euforic** [comment]
Nope just iterm2 and then directly into lima shell

**Comment by afbjorklund** [comment]
I was thinking that it would have the usual "select text" and copy/paste functions, but am missing something...

**Comment by jandubois** [comment]
> I was thinking that it would have the usual "select text" and copy/paste functions, but am missing something...

It does, and it works fine if you use iTerm copy/paste functionality.

@euforic Are you using vim commands to copy into the clipboard? That will not work because the Linux vim is not connected to the macOS clipboard. But if you select the text using iTerm and copy via ⌘C, then it works fine.

**Comment by afbjorklund** [comment]
This plugin looked interesting: https://github.com/ojroques/vim-oscyank

**Comment by euforic** [comment]
I will dig a little deeper into my config make sure its not something on my end. Thank you for the help 🤙🏽. Also thank you for this project it is awesome. All the work is truly appreciated 🙏🏼

**Comment by euforic** [comment]
> > I was thinking that it would have the usual "select text" and copy/paste functions, but am missing something...
> 
> It does, and it works fine if you use iTerm copy/paste functionality.
> 
> @euforic Are you using vim commands to copy into the clipboard? That will not work because the Linux vim is not connected to the macOS clipboard. But if you select the text using iTerm and copy via ⌘C, then it works fine.

Yeah I am using the system clipboard and was trying to bridge the linux system clipboard with then Mac OS system clip board similar to how [UTM](https://github.com/utmapp/UTM) does it with the Spice tools

**Comment by afbjorklund** [comment]
https://iterm2.com/documentation-utilities.html - see it2copy

Copies text to the pasteboard. Works over ssh. Accepts either standard input or a named file.

**Comment by ryuheechul** [comment]
It turns out (and TIL), sharing clipboard isn't a straight forward thing to make it happen (at least via SSH, X11 forwarding seems to be the promising way to do so).
So I was able to [make it work](https://gist.github.com/ryuheechul/9515381570a0cea994e62647d92a864f#sharing-clipboard) via that way.

This requires installing XQuartz and X11 forwarding (which is described in the gist) and I chose to SSH to lima guest via `ssh` (but this requires additional authentication setup - like adding public key to `authorized_keys` on lima)  instead of `lima` since that's the way I know at the moment. But lima configuration yaml also seems to support ssh configuration there, so you might get lucky with that.

Although it's not an auto magic way that lima just comes with, but I'm quite satisfied with the way it works at the moment.
